modules/home/transformations/filter-sidebar-upgrade-by-license.php

```diff
@@ -2,45 +2,47 @@
 namespace elementor\modules\home\transformations;
 
 use Elementor\Modules\Home\Transformations\Base\Transformations_Abstract;
+use Elementor\Core\Common\Modules\Connect\Module as ConnectModule;
 use Elementor\Utils;
 
-
 if ( ! defined( 'ABSPATH' ) ) {
 	exit; // Exit if accessed directly.
 }
 
 class Filter_Sidebar_Upgrade_By_License extends Transformations_Abstract {
-	public bool $has_pro;
-
-	public function __construct( $args ) {
-		parent::__construct( $args );
-
-		$this->has_pro = Utils::has_pro();
-	}
-
-	private function is_valid_item( $item ) {
-		$has_pro_json_not_free = $this->has_pro && 'pro' === $item['license'][0];
-		$is_not_pro_json_not_pro = ! $this->has_pro && 'free' === $item['license'][0];
-		$should_show = ! isset( $item['show'] ) || 'true' === $item['show'];
-		return ( $has_pro_json_not_free && $should_show ) || ( $is_not_pro_json_not_pro && $should_show );
-	}
 
 	public function transform( array $home_screen_data ): array {
 		$new_sidebar_upgrade = [];
+		$user_tier = $this->get_tier();
 
-		foreach ( $home_screen_data['sidebar_upgrade'] as $index => $item ) {
-			if ( $this->is_valid_item( $item ) ) {
+		array_filter( $home_screen_data['sidebar_upgrade'], function( $item ) use ( &$new_sidebar_upgrade, $user_tier ) {
```

Comment:
I would avoid using reference `&` 
It is a bit old...
https://schlueters.de/blog/archives/125-Do-not-use-PHP-references.html

Can we actually return a value?